precompiles/common/expected_keepers.go

```diff
@@ -43,3 +44,13 @@ type StakingKeeper interface {
 	BeginRedelegate(goCtx context.Context, msg *stakingtypes.MsgBeginRedelegate) (*stakingtypes.MsgBeginRedelegateResponse, error)
 	Undelegate(goCtx context.Context, msg *stakingtypes.MsgUndelegate) (*stakingtypes.MsgUndelegateResponse, error)
 }
+
+type GovKeeper interface {
+	AddVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAddress, options govtypes.WeightedVoteOptions) error
```

Comment:
do we want to add `submit-proposal`?

Reply:
do we currently allow users to submit proposal on UI on cosmos side? (feel like this precompile is only relevant for users interacting with the chain through evm UI)